src/App.tsx

```diff
@@ -15,48 +16,102 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  ALPHABETICAL = 'A',
+  LENGTH = 'L',
+}
+
+const IS_SORTED_ALPHABET:SortType = SortType.ALPHABETICAL;
+const IS_SORTED_LENGTH:SortType = SortType.LENGTH;
+
+function getPreparedGoods(
+  goods: string[],
+  isSorted: string | '',
+  isReversed: boolean,
+): string[] {
+  const preparedGoods = [...goods];
+
+  switch (isSorted) {
+    case IS_SORTED_ALPHABET:
+      preparedGoods.sort((good1, good2) => good1.localeCompare(good2));
+      break;
+
+    case IS_SORTED_LENGTH:
+      preparedGoods.sort((good1, good2) => good1.length - good2.length);
+      break;
+
+    default:
+
+      break;
+  }
+
+  if (isReversed) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
 export const App: React.FC = () => {
+  const [isSorted, setIsSorted] = useState('');
```

Comment:
`isSorted` sounds like it has a boolean value...change this name to `isSortedBy` or `sortField`...

don't forget to rename setState function as well